pkg/syncer/pv_to_vvol_mapping.go

```diff
@@ -0,0 +1,157 @@
+		}
+	}
+
+	// pv to vvol Map maps vol.VolumeId.Id to backingobjectId.
+	volumeIdToBackingObjectIdMap := make(volumeIdHealthStatusMap, len(queryAllResult.Volumes))
+
+	for _, vol := range queryAllResult.Volumes {
+		volumeIdToBackingObjectIdMap[vol.VolumeId.Id] = vol.BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails).BackingDiskId
+	}
+
+	for volID, pvc := range volumeHandleToPvcMap {
+		pvTovvolPair := volID + ":" + volumeIdToBackingObjectIdMap[volID]
+		updatePVtoVvolMappingStatus(ctx, k8sclient, pvc, pvTovvolPair)
+	}
+
+	log.Info("csiGetPVtoVvolMapping: end")
+}
+
+func parseMappingString(pvTovvolPair string) (string, string, error) {
+	log := logger.GetLogger(ctx)
+	split := strings.Split(pvTovvolPair, ":")
+	if len(split) != 2 {
+		log.Error("Failed to parse mapping string")
+		return "", "", errors.New("Failed to parse mapping string")
+	}
+	return split[0], split[1], nil
+}
+
+func updatePVtoVvolMappingStatus(ctx context.Context, k8sclient clientset.Interface,
+	pvc *v1.PersistentVolumeClaim, pvTovvolPair string) {
+	log := logger.GetLogger(ctx)
+
+	uuid, backingDiskObjectId, err := parseMappingString(pvTovvolPair)
+	if err != nil {
+		log.Errorf("updatePVtoVvolMappingStatus: Failed to parse mapping pair for pvc %s/%s with err:%+v",
+			pvc.Namespace, pvc.Name, err)
+	}
+	val, found := pvc.Annotations[annPVtoVVOL+ "/" + uuid]
+
+	if !found || val != backingDiskObjectId {
+		// VolumeHealth annotation on pvc is changed, set it to new value.
+		metav1.SetMetaDataAnnotation(&pvc.ObjectMeta, annPVtoVVOL+ "/" + uuid, backingObjectId)
+
+		log.Infof("updatePVtoVvolMappingStatus: set pv to vvol annotation for pvc %s/%s from old "+
+			"value %s to new value %s",
+			pvc.Namespace, pvc.Name, val, backingDiskObjectId)
+		_, err := k8sclient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Update(ctx, pvc, metav1.UpdateOptions{})
+		if err != nil {
+			if apierrors.IsConflict(err) {
+				log.Debugf("updatePVtoVvolMappingStatus: Failed to update pvc %s/%s with err:%+v, will retry the update",
+					pvc.Namespace, pvc.Name, err)
+				// pvc get from pvcLister may be stale, try to get updated pvc which
+				// bound to pv from API server.
+				newPvc, err := k8sclient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(
+					ctx, pvc.Name, metav1.GetOptions{})
+				if err == nil {
+					log.Infof("updatePVtoVvolMappingStatus: updating pv to vvol annotation for pvc %s/%s which "+
+						"get from API server from old value %s to new value %s",
+						newPvc.Namespace, newPvc.Name, val, backingDiskObjectId)
+					metav1.SetMetaDataAnnotation(&newPvc.ObjectMeta, annPVtoVVOL+ "/" + uuid, backingDiskObjectId)
+					_, err := k8sclient.CoreV1().PersistentVolumeClaims(newPvc.Namespace).Update(ctx,
+						newPvc, metav1.UpdateOptions{})
+					if err != nil {
+						log.Errorf("updatePVtoVvolMappingStatus: Failed to update pvc %s/%s with err:%+v",
+							newPvc.Namespace, newPvc.Name, err)
+					}
```

Comment:
This block is a bit weird, I think this is the convention is to check if `if err!=nil`, if there is no error there should be your code that attempts to update the PVC again.